src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs

```diff
@@ -300,40 +394,59 @@ public override async ValueTask<FormulaValue> Visit(CallNode node, EvalVisitorCo
                     args[i] = await child.Accept(this, context.IncrementStackDepthCounter()).ConfigureAwait(false);
                 }
                 else
-                {
+                {
+                    // This is where Lambdas are created. They close over key values to invoke.
                     args[i] = new LambdaFormulaValue(node.IRContext, child, this, context);
                 }
             }
 
             var childContext = context.SymbolContext.WithScope(node.Scope);
 
             FormulaValue result;
+
+            // Remove this: https://github.com/microsoft/Power-Fx/issues/2821
             IReadOnlyDictionary<TexlFunction, IAsyncTexlFunction> extraFunctions = _services.GetService<IReadOnlyDictionary<TexlFunction, IAsyncTexlFunction>>();

```

Comment:
>extraFunctions = _services.GetService<IReadOnlyDictionary<TexlFunction, IAsyncTexlFunction>>(); [](http://example.com/codeflow?start=66&length=95)

Were you trying to address this problem in this PR?

Reply:
No - that was out of scope, but I filed #2821 to track it.